mucommander-core/src/main/java/com/mucommander/ui/action/impl/ShowTerminalAction.java

```diff
@@ -0,0 +1,140 @@
+import com.mucommander.core.desktop.DesktopManager;
+import com.mucommander.text.Translator;
+import com.mucommander.ui.action.AbstractActionDescriptor;
+import com.mucommander.ui.action.ActionCategory;
+import com.mucommander.ui.action.ActionDescriptor;
+import com.mucommander.ui.action.ActionProperties;
+import com.mucommander.ui.main.MainFrame;
+
+import java.awt.BorderLayout;
+import java.awt.event.KeyAdapter;
+import java.awt.event.KeyEvent;
+import java.util.Map;
+
+import javax.swing.KeyStroke;
+import javax.swing.SwingUtilities;
+
+/**
+ * This action shows built-in terminal (it mimics the behavior of Midnight Commander Ctrl-O command
+ * that originates back from Norton Commander).
+ */
+public class ShowTerminalAction extends ActiveTabAction {
+
+    private JediTermWidget terminal;
+
+    public ShowTerminalAction(MainFrame mainFrame, Map<String, Object> properties) {
+        super(mainFrame, properties);
+
+        setEnabled(DesktopManager.canOpenInFileManager());
+    }
+
+    @Override
+    protected void toggleEnabledState() {
+        AbstractFile currentFolder = mainFrame.getActivePanel().getCurrentFolder();
+        setEnabled(currentFolder.getURL().getScheme().equals(LocalFile.SCHEMA));
+    }
+
+    @Override
+    public void performAction() {
+        if (terminal == null) {
+            try {
+                LOGGER.info("Going to show Terminal...");
+                mainFrame.getSplitPane().setVisible(false);
+                terminal = getTerminal();
+                mainFrame.getMainPanel().add(terminal, BorderLayout.CENTER);
+                terminal.revalidate();
+                SwingUtilities.invokeLater(() -> {
+                    terminal.requestFocusInWindow();
+                });
+
+                terminal.getTerminalPanel().addCustomKeyListener(new KeyAdapter() {
+                    public void keyPressed(KeyEvent keyEvent) {
+                        // TODO get the real configured key seq
+                        if (keyEvent.getKeyCode() == KeyEvent.VK_O &&
+                                (keyEvent.getModifiersEx() & KeyEvent.CTRL_DOWN_MASK) != 0) {
+                            revertToTableView();
+                        }
+                    }
+                });
+            } catch (Exception e) {
+                LOGGER.error("Caught exception while trying to show Terminal", e);
+                revertToTableView();
+            }
+        } else {
+            revertToTableView();
+            terminal = null;
+        }
+
+    }
+
+    @Override
+    public ActionDescriptor getDescriptor() {
+        return new Descriptor();
+    }
+
+    public static class Descriptor extends AbstractActionDescriptor {
+        public static final String ACTION_ID = "ShowTerminal";
+
+        public String getId() {
+            return ACTION_ID;
+        }
+
+        public ActionCategory getCategory() {
+            return ActionCategory.NAVIGATION;
```

Comment:
`navigation` is probably not the right category for this
'Run command' was placed in 'File operations' but that doesn't sound right either, maybe it should be 'misc'?